test/e2e/_macros.js

```diff
@@ -7,291 +7,208 @@
 import * as cp from "node:child_process";
 
 import test from "ava";
+import isCI from "is-ci";
 
-import * as constants from "../_constants.cjs";
+/* eslint-disable ava/no-import-test-files */
+import * as execFileTest from "../fuzz/exec-file.test.cjs";
+import * as execTest from "../fuzz/exec.test.cjs";
+import * as forkTest from "../fuzz/fork.test.cjs";
+import * as spawnTest from "../fuzz/spawn.test.cjs";
+/* eslint-enable ava/no-import-test-files */
 
-import * as shescape from "../../index.js";
+/**
+ * Converts a test function into a test function that catches and discards some
+ * errors that are allowed during testing.
+ *
+ * In particular:
+ * - Allows for ENOENT errors when run outside a CI. This allows for running the
+ * e2e tests locally even if you don't have all shells installed.
+ *
+ * @param {Function} cb The function to run and handle errors for.
+ * @returns {Function} A callback that will safely handle some errors.
+ */
+const tryRun = (cb) => {
+  const handleError = (error) => {
+    if (!isCI) {
+      if (error.code === "ENOENT" || error.toString().includes("ENOENT")) {
+        return;
+      }
+    }
+
+    throw error;
+  };
```

Comment:
It is a bit unfortunate this makes it so tests are (effectively) [`skip`ped](https://github.com/avajs/ava/blob/b208d143ad852dc95aa8b44eed94ac1f404a25f4/docs/01-writing-tests.md#skipping-tests) silently. On second thought, think it would be better if we check for shell availability explicitly and skip tests explicitly so the person running the tests can actually see these tests being skipped.

Reply:
Resolved with d5ce41426555c0c2ce60354415fff85477621338